### PR TITLE
Fix empty data check when merging v2 node responses

### PR DESCRIPTION
**Pull request by shifucun**
When no data is present, the empty data is not nil but rather empty arcs and nodes.

### CODE REVIEW

File: internal/merger/merger_test.go [hunk 1]
```diff
@@ -226,7 +226,15 @@ func TestMergeNode(t *testing.T) {
 			},
 		},
 		{
-			&pbv2.NodeResponse{},
+			&pbv2.NodeResponse{
+				Data: map[string]*pbv2.LinkedGraph{
+					"dcid1": {
+						Arcs: map[string]*pbv2.Nodes{
+							"prop1.2": {},
```
Comment (keyurva) [inline internal/merger/merger_test.go:233]: Perhaps also add a test case where the value is `nil`?

```
Arcs: map[string]*pbv2.Nodes{
  "prop1.2": nil,
}
```

**Review by keyurva — APPROVED** [review]
Thanks for the fix.

File: internal/merger/merger_test.go (hunk 1 above)
Comment (shifucun) [reply internal/merger/merger_test.go:273]: Done
